frame/glutton/src/lib.rs

```diff
@@ -0,0 +1,290 @@
+
+	/// Storage map used for wasting proof size.
+	///
+	/// It contains no meaningful data - hence the name "Trash". The maximal number of entries is
+	/// set to 65k, which is just below the next jump at 16^4. This is important to reduce the proof
+	/// size benchmarking overestimate. The assumption here is that we won't have more than 65k *
+	/// 1KiB = 65MiB of proof size wasting in practice. However, this limit is not enforces, so the
+	/// pallet would also work out of the box with more entries, but its benchmarked proof weight
+	/// would possibly be underestimated in that case.
+	#[pallet::storage]
+	pub(super) type TrashData<T: Config> = StorageMap<
+		Hasher = Twox64Concat,
+		Key = u32,
+		Value = [u8; 1024],
+		QueryKind = OptionQuery,
+		MaxValues = ConstU32<65_000>,
+	>;
+
+	/// The current number of entries in `TrashData`.
+	#[pallet::storage]
+	pub(crate) type TrashDataCount<T: Config> = StorageValue<_, u32, ValueQuery>;
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn integrity_test() {
+			assert!(
+				!T::WeightInfo::waste_ref_time_iter(1).ref_time().is_zero(),
+				"Weight zero; would get stuck in an infinite loop"
+			);
+			assert!(
+				!T::WeightInfo::waste_proof_size_some(1).proof_size().is_zero(),
+				"Weight zero; would get stuck in an infinite loop"
+			);
+		}
+
+		fn on_idle(_: BlockNumberFor<T>, remaining_weight: Weight) -> Weight {
+			let mut meter = WeightMeter::from_limit(remaining_weight);
+			if !meter.check_accrue(T::WeightInfo::empty_on_idle()) {
+				return T::WeightInfo::empty_on_idle()
+			}
+
+			let proof_size_limit = Storage::<T>::get().mul_floor(meter.remaining().proof_size());
+			let computation_weight_limit =
+				Compute::<T>::get().mul_floor(meter.remaining().ref_time());
+			let mut meter = WeightMeter::from_limit(Weight::from_parts(
+				computation_weight_limit,
+				proof_size_limit,
+			));
+
+			Self::waste_at_most_proof_size(&mut meter);
+			Self::waste_at_most_ref_time(&mut meter);
+
+			meter.consumed
+		}
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Initializes the pallet by writing into `TrashData`.
+		///
+		/// Only callable by Root. A good default for `trash_count` is `5_000`.
+		#[pallet::call_index(0)]
+		#[pallet::weight(
+			T::WeightInfo::initialize_pallet_grow(witness_count.unwrap_or_default())
+				.max(T::WeightInfo::initialize_pallet_shrink(witness_count.unwrap_or_default()))
+		)]
+		pub fn initialize_pallet(
+			origin: OriginFor<T>,
+			new_count: u32,
+			witness_count: Option<u32>,
+		) -> DispatchResult {
+			ensure_root(origin)?;
+
+			let current_count = TrashDataCount::<T>::get();
+			ensure!(
+				current_count == witness_count.unwrap_or_default(),
+				Error::<T>::AlreadyInitialized
+			);
+
+			// Add or remove elements from `TrashData`.
+			(new_count..current_count).for_each(|i| TrashData::<T>::remove(i));
+			(current_count..new_count).for_each(|i| TrashData::<T>::insert(i, [i as u8; 1024]));
```

Comment:
Purely a stylistic nitpick: this is a little confusing on the first glance; I think just putting an explicit `if` would be more readable (even if the current code is *functionally* fine it looks a little weird), e.g.:

```suggestion
			if new_count > current_count {
				(current_count..new_count).for_each(|i| TrashData::<T>::insert(i, [i as u8; 1024]));
			} else {
				(new_count..current_count).for_each(TrashData::<T>::remove);
			}
```

Reply:
Haha, yea had the `if` there first, but then thought it looked cooler without :sunglasses:  
Anyway I will make it more readable for you :heart: